rest/route/middleware.go

```diff
@@ -34,6 +35,7 @@ type (
 const (
 	// These are private custom types to avoid key collisions.
 	RequestContext requestContextKey = 0
+	payloadKey     requestContextKey = 3
```

Comment:
nit: I would call this `githubPayloadKey` since payload is kind of generic (unless we think it'll be used elsewhere).